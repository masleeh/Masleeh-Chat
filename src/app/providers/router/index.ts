import AppRouter from "./ui/AppRouter";
import { AppRouterProvider, router as AppBrowserRouter } from "./ui/RouterProvider";

export {
    AppRouter,
    AppRouterProvider,
    AppBrowserRouter
}