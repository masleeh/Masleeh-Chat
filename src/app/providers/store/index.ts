import StoreProvider from "./StoreProvider";
import type { IStateSchema } from "./config/StateSchema";

export {
    StoreProvider
}

export type {
    IStateSchema
}