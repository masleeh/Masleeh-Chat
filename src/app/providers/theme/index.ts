import AppThemeProvider from './ui/ThemeProvider'

export {
    AppThemeProvider
}