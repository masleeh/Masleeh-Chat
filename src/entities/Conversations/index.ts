import { convActions, convReducer } from "./model/slice/conversations.slice";
import getConvsThunk from "./api/getConvsThunk/getConvsThunk";
import { getConvState } from "./model/selectors/getConvState/getConvState";
import { getConvData } from "./model/selectors/getConvData/getConvData";
import type {TConvItem} from './model/types/conversations.state'

export {
    convActions, 
    convReducer,
    getConvsThunk,
    getConvState,
    getConvData,
}

export type {
    TConvItem
}