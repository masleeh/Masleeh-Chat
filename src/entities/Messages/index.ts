import getMessagesThunk from "./api/getMessagesThunk/getMessagesThunk";
import { getMessages } from "./model/selectors/getMessages/getMessages";
import { getMessagesState } from "./model/selectors/getMessagesState/getMessagesState";
import { messagesActions, messagesReducer } from "./model/slice/messages.slice";
import type { IMessage, IMessagesSchema } from "./model/types/messages.state";

export {
    messagesActions,
    messagesReducer,
    getMessagesThunk,
    getMessagesState,
    getMessages
}

export type {
    IMessagesSchema,
    IMessage
}