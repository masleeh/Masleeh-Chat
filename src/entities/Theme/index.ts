import { getTheme } from './model/selectors/getTheme/getTheme'
import { getThemeTitle } from './model/selectors/getThemeTitle/getThemeTitle'
import { themeActions, themeReducer } from './model/slice/theme.slice'
import { AppThemes, ThemeMode, Themes } from './model/themes/themes'
import {IThemeSchema} from './model/types/theme.state'

export type {
    IThemeSchema,
    
}

export {
    getTheme,
    getThemeTitle,
    themeActions,
    themeReducer,
    AppThemes,
    ThemeMode,
    Themes
}