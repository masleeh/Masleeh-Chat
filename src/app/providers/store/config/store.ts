import { ReducersMapObject, configureStore } from "@reduxjs/toolkit"
import { IStateSchema } from "./StateSchema"
import { userReducer } from "entities/User"


export const createReduxStore = (initialState?: IStateSchema) => {
    const rootReducers: ReducersMapObject<IStateSchema> = {
        user: userReducer
    }

    return configureStore<IStateSchema>({
        reducer: rootReducers,
        devTools: true,
        preloadedState: initialState
    })
}

export type RootState = ReturnType<typeof createReduxStore>['getState']
export type AppDispatch = ReturnType<typeof createReduxStore>['dispatch']