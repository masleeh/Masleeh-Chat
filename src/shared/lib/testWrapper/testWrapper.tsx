import { render } from "@testing-library/react"
import { IStateSchema, TestStoreProvider } from "app/providers/store";
import i18nTest from 'shared/config/i18n/i18nTest'
import { I18nextProvider } from "react-i18next"
import { MemoryRouter } from "react-router-dom";
import { DeepPartial, EnhancedStore } from "@reduxjs/toolkit";
import { AppThemeProvider } from "app/providers/theme";

export interface componentRenderOptions {
    route?: string;
    initialState?: DeepPartial<IStateSchema>;
    store?: EnhancedStore
}

const testWrapper = (component: React.ReactNode, options: componentRenderOptions = {}) => {
    const {
        route = '/',
        initialState,
        store
    } = options

    return render(
        <TestStoreProvider initialState={initialState} store={store}>
            <AppThemeProvider>
                <MemoryRouter initialEntries={[route]}>
                    <I18nextProvider i18n={i18nTest}>
                        {component}
                    </I18nextProvider>
                </MemoryRouter>
            </AppThemeProvider>
        </TestStoreProvider>
    )
}

export default testWrapper