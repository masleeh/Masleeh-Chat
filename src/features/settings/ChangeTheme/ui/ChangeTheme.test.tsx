import { DeepPartial } from '@reduxjs/toolkit'
import {screen} from '@testing-library/react'
import { IStateSchema, createReduxStore } from 'app/providers/store'
import { AppThemes, ThemeMode, Themes } from 'entities/Theme'
import testWrapper from "shared/lib/testWrapper/testWrapper"
import { ChangeTheme } from '..'
import userEvent from '@testing-library/user-event'

const state: DeepPartial<IStateSchema> = {
    theme: {
        title: Themes.default,
        theme: AppThemes[Themes.default][ThemeMode.light]
    }
}

describe('ChangeTheme component', () => {
    it('Check ChangeTheme snapshot', () => {
        const component = testWrapper(<ChangeTheme />)
        expect(component).toMatchSnapshot()
    })

    it('Check changing theme', async () => {
        const store = createReduxStore(state as IStateSchema)
        testWrapper(<ChangeTheme />, {store: store})
        const themeBtn = screen.getByLabelText("theme btn Violet")
        userEvent.click(themeBtn)

        setTimeout(() => {
            const newState = store.getState()
            expect(newState.theme.title).toBe(Themes.violet)
            expect(newState.theme.theme).toEqual(AppThemes[Themes.violet][ThemeMode.light])
        }, 1000)
    })

})