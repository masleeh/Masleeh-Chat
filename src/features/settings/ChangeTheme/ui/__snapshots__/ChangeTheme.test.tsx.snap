// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChangeTheme component > Check ChangeTheme snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiStack-root css-ofxwhm-MuiStack-root"
      >
        <div
          aria-label="theme btn Default"
          class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault css-qzru0f-MuiAvatar-root"
          data-mui-internal-clone-element="true"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ColorLensIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
            />
          </svg>
        </div>
        <div
          aria-label="theme btn Violet"
          class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault css-5522rq-MuiAvatar-root"
          data-mui-internal-clone-element="true"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ColorLensIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
            />
          </svg>
        </div>
        <div
          aria-label="theme btn Feldgrau"
          class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault css-1y63tu3-MuiAvatar-root"
          data-mui-internal-clone-element="true"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ColorLensIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
            />
          </svg>
        </div>
        <div
          aria-label="theme btn Mindaro"
          class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault css-11yndga-MuiAvatar-root"
          data-mui-internal-clone-element="true"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ColorLensIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
            />
          </svg>
        </div>
        <div
          aria-label="theme btn Lightred"
          class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault css-6j7eok-MuiAvatar-root"
          data-mui-internal-clone-element="true"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ColorLensIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiStack-root css-ofxwhm-MuiStack-root"
    >
      <div
        aria-label="theme btn Default"
        class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault css-qzru0f-MuiAvatar-root"
        data-mui-internal-clone-element="true"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ColorLensIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
          />
        </svg>
      </div>
      <div
        aria-label="theme btn Violet"
        class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault css-5522rq-MuiAvatar-root"
        data-mui-internal-clone-element="true"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ColorLensIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
          />
        </svg>
      </div>
      <div
        aria-label="theme btn Feldgrau"
        class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault css-1y63tu3-MuiAvatar-root"
        data-mui-internal-clone-element="true"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ColorLensIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
          />
        </svg>
      </div>
      <div
        aria-label="theme btn Mindaro"
        class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault css-11yndga-MuiAvatar-root"
        data-mui-internal-clone-element="true"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ColorLensIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
          />
        </svg>
      </div>
      <div
        aria-label="theme btn Lightred"
        class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault css-6j7eok-MuiAvatar-root"
        data-mui-internal-clone-element="true"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ColorLensIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
